File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/front/status/list/PagerAdapter.java

package com.ikmr.banbara23.yaeyama_liner_checker.front.status.list;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;

import com.ikmr.banbara23.yaeyama_liner_checker.model.Company;

/**
 * PagerAdapter
 */
public class PagerAdapter extends FragmentStatePagerAdapter {
    int mNumOfTabs;

    public PagerAdapter(FragmentManager fm, int NumOfTabs) {
        super(fm);
        mNumOfTabs = NumOfTabs;
    }

    @Override
    public Fragment getItem(int position) {
        Company company;
        switch (position) {
            case 0:
                company = Company.ANEI;
                break;
            case 1:
                company = Company.YKF;
                break;
            case 2:
                company = Company.DREAM;
                break;
            default:
                company = null;
        }
        return StatusListTabFragment.NewInstance(company);
    }

    @Override
    public int getCount() {
        return mNumOfTabs;
    }

    public Fragment findFragmentByPosition(ViewPager viewPager, int position) {
        return (Fragment) instantiateItem(viewPager, position);
    }
}

File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/front/status/list/PagerAdapter.kt
package com.ikmr.banbara23.yaeyama_liner_checker.front.status.list

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.ikmr.banbara23.yaeyama_liner_checker.model.Company

/**
 * 会社一覧タブ PagerAdapter
 */
class PagerAdapter(
    fm: FragmentManager,
    private var tabCount: Int
) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment {
        val company = when (position) {
            0 -> Company.ANEI
            1 -> Company.YKF
            else -> Company.ANEI
        }
        return StatusListTabFragment.NewInstance(company)
    }

    override fun getPageTitle(position: Int): String {
        return if (position == 0) "安栄観光" else "八重山観光\nフェリー"
    }

    override fun getCount(): Int {
        return tabCount
    }
}
